collectors/ebpf.plugin/ebpf_hardirq.c

```diff
@@ -0,0 +1,503 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "ebpf.h"
+#include "ebpf_hardirq.h"
+
+struct config hardirq_config = { .first_section = NULL,
+    .last_section = NULL,
+    .mutex = NETDATA_MUTEX_INITIALIZER,
+    .index = { .avl_tree = { .root = NULL, .compar = appconfig_section_compare },
+        .rwlock = AVL_LOCK_INITIALIZER } };
+
+#define HARDIRQ_MAP_LATENCY 0
+#define HARDIRQ_MAP_LATENCY_STATIC 1
+static ebpf_local_maps_t hardirq_maps[] = {
+    {
+        .name = "tbl_hardirq",
+        .internal_input = NETDATA_HARDIRQ_MAX_IRQS,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_STATIC,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    },
+    {
+        .name = "tbl_hardirq_static",
+        .internal_input = HARDIRQ_EBPF_STATIC_END,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_STATIC,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    },
+    /* end */
+    {
+        .name = NULL,
+        .internal_input = 0,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_CONTROLLER,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    }
+};
+
+static ebpf_data_t hardirq_data;
+
+static ebpf_tracepoint_t hardirq_tracepoints[] = {
```

Comment:
When we have a big sample of repeated strings like we have here, old compilers could create a lot of garbage, I suggest to create definitions for `irq` and 'irq_vectors` and use these defines here.

Reply:
Do you mean to use the C preprocessor? `#define HARDIRQ_TP_CLASS_IRQ "irq"`? Will this not just paste the same exact string and be effectively the same thing?

Or do you mean to use `const char *`? That would likely guarantee deduplication in old compilers that behave like this.

Reply:
FYI I tried doing it with `const char *` but had way too many issues with the compiler complaining e.g. `"initializer element is not constant"` and other const-related things. It was easier to just leave it as is.

Reply:
> Do you mean to use the C preprocessor? #define HARDIRQ_TP_CLASS_IRQ "irq"? Will this not just paste the same exact string and be effectively the same thing?

Yes, I was thinking to use preprocessors, this is a pattern we are using in different plugins like [perf.plugin](https://github.com/netdata/netdata/blob/ade4b8798173142fd7cf70be949343217dbe72b0/collectors/perf.plugin/perf_plugin.c#L63-L66).

Reply:
Alright, addressed in https://github.com/netdata/netdata/pull/11410/commits/72f4700c1fd75d7132f9a38ebef5e0212d6d6f3a